sklearn/impute/_base.py

```diff
@@ -396,17 +402,29 @@ def transform(self, X):
                 indexes = np.repeat(np.arange(len(X.indptr) - 1, dtype=np.int),
                                     np.diff(X.indptr))[mask]
 
+                if self.add_indicator:
+                    sparse_constructor = (sparse.csr_matrix
```

Comment:
This logic is included three times. Can we create a helper for it, or just include it in _get_mask??

Reply:
Creating a function to do so, would be nice, if that okay.

Reply:
Yes, it's fine to make a helper function. Still would be nice if it can be included in the existing function